contracts/locked/LockedSOV.sol

```diff
@@ -276,17 +288,25 @@ contract LockedSOV {
 
 		require(cliff == vesting.cliff() && duration == vesting.duration(), "Wrong Vesting Schedule.");
 
-		_stakeTokens(address(vesting));
+		_stakeTokens(msg.sender, address(vesting));
 	}
 
 	/**
 	 * @notice Withdraws unlocked tokens and Stakes Locked tokens for a user who already have a vesting created.
 	 * @param _receiverAddress If specified, the unlocked balance will go to this address, else to msg.sender.
-	 * @dev The user should already have a vesting created, else this function will throw error.
 	 */
 	function withdrawAndStakeTokens(address _receiverAddress) external {
-		withdraw(_receiverAddress);
-		stakeTokens();
+		_withdraw(msg.sender, _receiverAddress);
+		_createVestingAndStake(msg.sender);
+	}
+
+	/**
+	 * @notice Withdraws unlocked tokens and Stakes Locked tokens for a user who already have a vesting created.
+	 * @param _userAddress The address of user tokens will be withdrawn.
+	 */
+	function withdrawAndStakeTokensFrom(address _userAddress) external {
+		_withdraw(_userAddress, _userAddress);
+		_createVestingAndStake(_userAddress);
 	}
 
 	/**
```

Comment:
The idea is to make only the governance as the admin, so any change will be done through SIPs. Also, even though the admin makes the change, the user is still in control of their funds. But I understand the issue you mentioned. Maybe @ororopickpocket has some comments?

Reply:
if we would have only 1 admin it would be clearer for everyone that there are not several admins on the system with each of them having such power. but i do not think, we have to change it, as long as we are aware that the only admin is the governance (though then we could have just used the owner role here, I suppose).